apis/v1alpha1/clusterprofile_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterProfileSpec defines the desired state of ClusterProfile.
+type ClusterProfileSpec struct {
+	// DisplayName defines a human-readable name of the ClusterProfile
+	// +optional
+	DisplayName string `json:"displayName"`
+
+	// ClusterManager defines which cluster manager owns this ClusterProfile resource
```

Comment:
more than own, its to self-identify the "creator" of the ClusterInventory object, not necessarily what/who manages it?
Does it replace the "Owner" field of the resource?

Reply:
It doesn't replace the Owner field. This field is to indicate which cluster manager this resource came from so that if it's an environment with multiple cluster managers, then each manager knows which resources to manage.
I had created in the beginning but it was suggested to change to own: https://github.com/kubernetes-sigs/cluster-inventory-api/pull/6#discussion_r1568165541
WDYT? How should I best describe this?

Reply:
Thank you for the details.

I see you added a bit more details on ClusterManager. I think I understand better your intent.
Now, it sounds like this field is only a hint for the managers to not step on each other (I don't think it will be permission protected right?). If we assume it's only a "hint", why keep it immutable and required?

Could a cluster manager transfer management to another one (or an admin deciding it wants to transition it?), Could we have a situation where the clsuter profile is manually created and therefore has no manager?
If we push it in this direction (optional and mutable), why not just using the label?

Does knowing what the manager is convey information to consumers of the ClusterProfile object?

Reply:
I am not worried about changing the cluster manager. One could argue that the ClusterProfile should be deleted and recreated in order to change the manager. In that case ClusterManager can remain immutable. What worries me more is that we allow only one ClusterManager for each member cluster. Having only one manager simplifies certain operations, but I wonder if it is too limiting. What if various operations of a fleet are performed by multiple ClusterManagers?

Reply:
> What if various operations of a fleet are performed by multiple ClusterManagers?

ClusterProfile is meant to be continuously updated by one ClusterManager. If we allow multiple ClusterManagers to update it then I think it will be hard to coordinate who is allow to update which fields without stepping on each other like Corentin said.

> One could argue that the ClusterProfile should be deleted and recreated in order to change the manager. In that case ClusterManager can remain immutable.

This is a great point. If we can come to a conclusion on this I will add it to the field description.

Reply:
> I don't think it will be permission protected right?

We do have the CEL validation for now to block it from being updated.

> If we assume it's only a "hint", why keep it immutable and required?

I think it's important to prevent multiple ClusterManagers from updating the same ClusterProfile so consumers can get a consistent feed of data from one source. 

> Could a cluster manager transfer management to another one

I like Babak comment above regarding delete and recreate but I hope others can chime in on this as well. CC @qiujian16 @dixudx 

> Does knowing what the manager is convey information to consumers of the ClusterProfile object?

This is a good point. As you suggested, this information seems to be only useful for the producers but I am thinking it might be useful in some cases for the consumer as well. For example, maybe the consumer can be the provider as well. Maybe ArgoCD can have it's own cluster inventory which produces ClusterProfile objects but also have the mechanism to import ClusterProfiles into its own registry. In that case, it might be helpful to know where the ClusterProfile it came from? See https://github.com/argoproj-labs/argocd-operator/issues/957

Reply:
Assuming that changing the cluster manager is a rare event, keeping it immutable is a good idea. That way the cluster managers know that the list of their managed clusters won't change and they only need to watch delete/create events to update the list.

Reply:
Thank you for the long answer.
CEL validation for the label is good.

Explanation and extra details for clusterManager helped. I think they would "own" the object since it's immutable, but the result is the same.
Immutable is good and possible to relax (while going the other way is not possible).

For specific parameters where we fear overlap, we can have additional rules (kinda similar to scheduler owning a subset of fields).

Change looks good for me :)

Reply:
> If we allow multiple ClusterManagers to update it then I think it will be hard to coordinate who is allow to update which fields without stepping on each other like Corentin said.

If there's a need to support multiple controllers updating one ClusterProfile resource, we could look to the pattern Gateway API uses for [RouteStatus](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.RouteParentStatus), with a separate set of status conditions scoped to each controller. I'm not sure if this capability is actually needed though, or if it would be better to keep the ClusterManager relationship simpler.

Reply:
I don't think we restrict one ClusterManager to have only one controller (i.e. we can have one controller collect cpu/memory property while another one collect ARD/Device property in the same clusterManager). However, I don't know if the benefit of supporting multiple ClusterManager for a single ClusterProfile worth breaking the neat immutability of the clusterManager field.

Reply:
Resolving this for now. We defined what a cluster manager is and decided that it should be an immutable field for now.